src/mono/mono/metadata/sysmath.c

```diff
@@ -109,78 +109,66 @@ ves_icall_System_Math_Acosh (gdouble x)
 	return acosh (x);
 }
 
-gdouble 
+gdouble
 ves_icall_System_Math_Asin (gdouble x)
 {
 	return asin (x);
 }
 
-gdouble 
+gdouble
 ves_icall_System_Math_Asinh (gdouble x)
 {
 	return asinh (x);
 }
 
-gdouble 
+gdouble
 ves_icall_System_Math_Atan (gdouble x)
 {
 	return atan (x);
 }
 
-gdouble 
+gdouble
 ves_icall_System_Math_Atan2 (gdouble y, gdouble x)
 {
 	return atan2 (y, x);
 }
 
-gdouble 
+gdouble
 ves_icall_System_Math_Atanh (gdouble x)
 {
 	return atanh (x);
 }
 
-gdouble 
+gdouble
 ves_icall_System_Math_Exp (gdouble x)
 {
 	return exp (x);
 }
 
-gdouble 
+gdouble
 ves_icall_System_Math_Log (gdouble x)
 {
 	return log (x);
 }
 
-gdouble 
+gdouble
 ves_icall_System_Math_Log10 (gdouble x)
 {
 	return log10 (x);
 }
 
-gdouble 
+gdouble
 ves_icall_System_Math_Pow (gdouble x, gdouble y)
 {
 	return pow (x, y);
 }
 
-gdouble 
+gdouble
 ves_icall_System_Math_Sqrt (gdouble x)
 {
 	return sqrt (x);
 }
 
-gdouble
-ves_icall_System_Math_Abs_double (gdouble v)
-{
-	return fabs (v);
```

Comment:
For mono it's basically `vandps  xmm0, xmm0, XMMWORD PTR .LC0[rip]`
while you replace it with GPR based impl. However, you get rid of a on Internal Call so it's probably better still

Reply:
https://godbolt.org/z/xEnzqharx

Reply:
With clang, there is no difference: https://godbolt.org/z/Mhj1EfbPa

Reply:
ah oops, it was gcc 😄